crates/namada/src/ledger/native_vp/masp.rs

```diff
@@ -430,19 +366,12 @@ where
     // Check if IBC message was received successfully in this state transition
     fn is_receiving_success(
         &self,
-        packet: &Packet,
-        keys_changed: &BTreeSet<Key>,
+        src_port_id: &PortId,
+        src_channel_id: &ChannelId,
+        sequence: Sequence,
     ) -> Result<bool> {
         // Ensure that the event corresponds to the current changes to storage
-        let ack_key = storage::ack_key(
-            &packet.port_id_on_a,
-            &packet.chan_id_on_a,
-            packet.seq_on_a,
-        );
-        if !keys_changed.contains(&ack_key) {
```

Comment:
Putting aside assumptions about the behaviour/judgement of the IBC VP, is there a chance that an IBC packet that has been previously acknowledged in a previous state change could be replayed here successfully and be used to fool the MASP VP into, say, reshielding funds from the packet?

Reply:
it is possible if IBC VP isn't triggered. I'll revert this.